services/order.py

```diff
@@ -0,0 +1,35 @@
+from django.db import transaction
+
+from db.models import Ticket, User, Order, MovieSession
+from django.db.models import DateTimeField, QuerySet
+
+
+@transaction.atomic
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: DateTimeField = None
+) -> None:
+    user = User.objects.get(username=username)
+    order = Order.objects.create(user=user)
+
+    if date:
+        order.created_at = date
+        order.save()
+
+    Ticket.objects.bulk_create([
+        Ticket(
+            movie_session=MovieSession.objects.get(id=ticket["movie_session"]),

```

Comment:
Use `movie_session_id` argument and pass the id directly.

Reply:
ya this thing doesnt work which bulk_create, i dont know why, but if I change bulk_create to just loop all ok

Reply:
But if you use `.get()` every time, this solution won't gain anything from `bulk_create()`. It will access the database for every ticket even before it executes.

Reply:
ya i know , i changed it already , but my electricity is gone XD

Reply:
I hope you saved your files before it went out!